sandbox/script.Rmd

```diff
@@ -0,0 +1,207 @@
+
+---
+title: "script checking"
+output: learnr::tutorial
+runtime: shiny_prerendered
+---
+
+<!-- New name! mentr -->
+
+```{r setup, include=FALSE}
+library(learnr)
+
+library(gradethis)
+# gradethis::grade_this_learnr()
+# learnr::tutorial_options(exercise.checker = grade_this_learnr)
+```
+
+### Script Style
+
+```{r barret0, exercise = TRUE, exercise.checker = grade_this_learnr}
+4
+```
+
+```{r barret0-solution}
+42
+```
+
+```{r barret0-check}
+expected <- 4
+
+grade_this({
+  if (.result == expected) pass("equals 4!")
+
+  if (identical(.result, expected)) pass("equals 4!")
+  testthat::expect_equal(.result, expected)
+
+  # pass_if_equal(4, MESSAGE)
+  # fail_if_equal(4, MESSAGE)
+
+  fail(random_encourage())
+})
```

Comment:
Should we try to differentiate between evaluation errors and error conditions thrown by testthat, etc

Reply:
We are able to detect differences as `testthat::expect(FALSE)` signals a `failure` class in addition to an `error` class.

Am worried about things like `checkmate` who eventually call `stop()` (via `checkmate::mstop()` via `checkmate::makeAssertion()`). There would be no way to differentiate those checks from actual errors.